File: JavaScript/Searching Algorithms/naiveStringSearch.js
function naiveStringSearch(str, pattern) {
    var counter = 0;
    for(var i = 0; i < str.length; i++){
        for(var j = 0; j < pattern.length; j++){
            if(pattern[j] !== str[i+j]) break;
            if(j === pattern.length - 1) counter++;
        }
    }
    return counter;
}

function naiveStringSearch2(str, pattern) {
    var len = pattern.length;
    var counter = 0;
    for(var i = 0; i <= str.length - pattern.length ; i++){
        if (str.slice(i,i+len) === pattern) {
            counter++;
        }
    }
    return counter;
}

console.log(naiveStringSearch("hahaha I am a haha  haha  haha boy","haha"));
console.log(naiveStringSearch("hahahahahaha I am a boy","haha"));
console.log(naiveStringSearch("hahaha I am a haha boy","boy"));


console.log(naiveStringSearch2("hahaha I am a haha  haha  haha boy","haha"));
console.log(naiveStringSearch2("hahahahahaha I am a boy","haha"));
console.log(naiveStringSearch2("hahaha I am a haha boy","boy"));

File: Python/Searching Algorithms/naiveStringSearch.py
def naiveStringSearch(str, pattern):
    counter = 0
    for i in range(0,len(str)):
        for j in range(0, len(pattern)):
            if pattern[j] != str[i+j]:
                break
            if j == len(pattern) - 1:
                counter += 1
    return counter

def stringSearch(str, pattern):
    counter = 0
    pattern_length = len(pattern)
    for i in range(0,len(str) - pattern_length + 1):
        if str[i:i+pattern_length] == pattern:
            counter += 1
    return counter

print(naiveStringSearch("hahaha I am a haha  haha  haha boy","haha"))
print(naiveStringSearch("hahahahahaha I am a boy","haha"))
print(naiveStringSearch("hahaha I am a haha boy","boy"))

print(stringSearch("hahaha I am a haha  haha  haha boy","haha"))
print(stringSearch("hahahahahaha I am a boy","haha"))
print(stringSearch("hahaha I am a haha boy","boy"))